tests/test_websocket.py

```diff
@@ -107,6 +107,27 @@ def test_build_request(websocket):
     )
 
 
+def test_build_request_custom_headers(websocket):
+    with pytest.raises(TypeError):
+        websocket.add_header(1, 2)
+    with pytest.raises(TypeError):
+        websocket.add_header(b'foo', 2)
+    websocket.add_header(b'foo', b'bar')
```

Comment:
These seem sensible to me as three tests, one for each invalid param, and one for the expected behaviour of a valid call. Not a blocker though.